anchore_engine/services/policy_engine/engine/feeds/client.py

```diff
@@ -333,193 +589,66 @@ def get_client(
-
-        while not success and count < retries:
-            count += 1
-            logger.debug("get attempt " + str(count) + " of " + str(retries))
-            try:
-                auth = (self.user, self.password)
-                logger.debug(
-                    "making authenticated request (user={}, conn_timeout={}, read_timeout={}, verify={}) to url {}".format(
-                        str(self.user), conn_timeout, read_timeout, verify, str(url)
-                    )
-                )
-                r = method(
-                    url, auth=auth, timeout=(conn_timeout, read_timeout), verify=verify
-                )
-                logger.debug("\tresponse status_code: " + str(r.status_code))
-                if r.status_code == 401:
-                    logger.debug(
-                        "Got HTTP 401 on authenticated {}, response body: {}".format(
-                            method.__name__, str(r.text)
-                        )
-                    )
-                    r.raise_for_status()
-                elif r.status_code == 200:
-                    success = True
-                    client_response.success = True
-                elif r.status_code in [403, 404]:
-                    r.raise_for_status()
-
-                client_response.status_code = r.status_code
-                client_response.content_type = r.headers["Content-Type"]
-                client_response.content = r.content
-                client_response.headers = r.headers
-            except requests.exceptions.ConnectTimeout as err:
-                logger.debug("attempt failed: " + str(err))
-                client_response.content = ensure_bytes(
-                    "server error: timed_out: " + str(err)
-                )
-                # return(ret)
-
-            except requests.HTTPError as e:
-                if e.response is not None and 400 <= e.response.status_code < 500:
-                    self._map_error_to_exception(e, username=self.user, url=url)
-                    # raise e
-                else:
-                    logger.debug("attempt failed: " + str(e))
-                    client_response.content = ensure_bytes("server error: " + str(e))
-            except Exception as err:
-                logger.debug("attempt failed: " + str(err))
-                client_response.content = ensure_bytes("server error: " + str(err))
-
-        return client_response
-
-
-class AnchoreIOClientError(AnchoreException):
-    def __init__(
-        self,
-        cause,
-        msg="Error initializing anchore.io client with configured credentials",
-    ):
-        self.cause = str(cause)
-        self.msg = msg
-
-    def __repr__(self):
-        return "{} - exception: {}".format(self.msg, self.cause)
+    unauthenticated_http_client = HTTPBasicAuthClient(
+        username=None,
+        password=None,
+        connect_timeout=conn_timeout,
+        read_timeout=read_timeout,
+        verify=verify,
+    )
```

Comment:
reordering of ops might be benefit this block. Fetch feeds_config inside the outer if block. grype_db_url could still be null dereferenced

Reply:
fixed